### DISCUSSION TITLE
How to FailWhenDeferred uder WiX4?

### DISCUSSION THREAD

**Discussion by oleg-shilo** [wixtoolset/issues#7602] [Questions] [answered]
The problem seems to be trivial. 
Previously (WiX3) it required `<CustomActionRef Id="FailWhenDeferred" />` from Util extension.

In WiX4 the documentation states that it has to be an element FailWhenDeferred.

![image](https://github.com/wixtoolset/issues/assets/16729806/de54255c-3bb2-43b7-83e3-5d554afbe979)

And yet te compiler error says that this element is unexpected to be a child of Package element:

```txt
d:\temp\setup.wxs(21) : error WIX0005: The Package element contains an unexpected child element 'FailWhenDeferred'.
``` 

I do include Util extension:
```
wix build  -ext WixToolset.Util.wixext setup.wxs -o setup.msi
```
Is documentation wrong or it is something else.

**Comment by robmen** [comment, marked as answer]
Not enough context from your code to answer fully. My best guess is that you did not add the `util` namespace. `FailWhenDeferred` is not part of the `wix` namespace.

**Comment by oleg-shilo** [comment]
Sorry, you are spot on.

I did add the XML namespace but forgot to use the prefix:


```xml
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util">
    . . .
    <FailWhenDeferred />
```

Changing it to `<util:FailWhenDeferred />` did the trick.

Thank you.
